dlt/destinations/path_utils.py

```diff
@@ -1,45 +1,225 @@
-# this can probably go some other place, but it is shared by destinations, so for now it is here
-from typing import List, Sequence, Tuple
+import re
+from typing import Any, Dict, List, Optional, Sequence, Set, Tuple
 
 import pendulum
-import re
 
-from dlt.destinations.exceptions import InvalidFilesystemLayout, CantExtractTablePrefix
+from dlt.cli import echo as fmt
+from dlt.common.storages.load_package import ParsedLoadJobFileName
+from dlt.destinations.exceptions import CantExtractTablePrefix, InvalidFilesystemLayout
+from dlt.destinations.impl.filesystem.typing import TCurrentDateTime
+
 
-# TODO: ensure layout only has supported placeholders
-SUPPORTED_PLACEHOLDERS = {"schema_name", "table_name", "load_id", "file_id", "ext", "curr_date"}
+SUPPORTED_PLACEHOLDERS = {
+    "schema_name",
+    "table_name",
+    "load_id",
+    "file_id",
+    "ext",
+    "curr_date",
+    "YYYY",
+    "Y",
+    "MMMM",  # January, February, March
+    "MMM",  # Jan, Feb, Mar
+    "MM",  # 01, 02, 03
+    "M",  # 1, 2, 3
+    "DD",  # 01, 02
+    "D",  # 1, 2
+    "HH",  # 00, 01, 02
+    "H",  # 0, 1, 2
+    "mm",
+    "m",
+    "dddd",  # Monday, Tuesday, Wednesday
+    "ddd",  # Mon, Tue, Wed
+    "dd",  # Mo, Tu, We
+    "d",  # 0-6
+    "Q",  # quarter 1, 2, 3, 4
+    "timestamp",
+}
 
 SUPPORTED_TABLE_NAME_PREFIX_PLACEHOLDERS = ("schema_name",)
 
 
-def check_layout(layout: str) -> List[str]:
+def get_placeholders(layout: str) -> List[str]:
+    return re.findall(r"\{(.*?)\}", layout)
+
+
+def prepare_datetime_params(
+    current_datetime: Optional[pendulum.DateTime] = None,
+    load_package_timestamp: Optional[str] = None,
+) -> Dict[str, str]:
+    # For formatting options please see
+    # https://github.com/sdispater/pendulum/blob/master/docs/docs/string_formatting.md
+    # Format curr_date datetime according to given format
+    params: Dict[str, str] = {}
+    now: pendulum.DateTime = current_datetime or pendulum.now()
+    # we need to preserve load package timestamp if it is given
+    # ideally it should always be given if not we take current datetime
```

Comment:
the opposite should be true. package datatime is our default. do not override what user is passing as current_datetime!

Reply:
@rudolfix I adjusted the logic, now is timestamp added if it has been passed.
And if `current_datetime` is not given then we take the current moment.